src/main.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::HashSet;
+use std::io::{Read, Write};
+use std::net::{TcpListener, TcpStream};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use rand::Rng;
+
+struct GameState {
+    secret_number: i32,
+    current_turn: usize,
+    attempts: Vec<usize>,
+    active: bool,
+}
+
+struct Lobby {
+    players: Vec<TcpStream>,
+}
+
+fn handle_client(mut stream: TcpStream, game_state: Arc<Mutex<GameState>>, player_id: usize, sent_messages: Arc<Mutex<HashSet<usize>>>) {
+    stream.write(b"Welcome to the game! The number is between 0 and 100.\n").expect("Failed to write to client");
+
+    let mut buffer = [0; 512];
+    loop {
+        {
+            let mut game_state = game_state.lock().unwrap();
+            if !game_state.active {
+                stream.write(b"Game Over: The game has ended.\n").expect("Failed to write to client");
+                return;
+            }
+
+            if game_state.current_turn != player_id {
+                stream.write(b"Waiting for another player to make a guess...\n").expect("Failed to write to client");
```

Comment:
pour l'aspect visuel du jeu peut être que ça serait bien de n'afficher qu'un seul message `"Waiting for another player to make a guess...\n"` parce que la ça va spam tant que c'est pas son tour. un ajout d'un booléen devrait faire l'affaire